### PR TITLE
Login & Signup

### PR DESCRIPTION
**Pull request by Wydder**
1. Created login and signup components;
2. Created auth guard to use it on pages where we will need to be authenticated with a user;
3. Created authentication API service (WIP - still need to add the correct POST URL when the backend is merged);
4. Created User model like the one in the backend project.

PS: Not yet fully tested, because of no backend URL.

### CODE REVIEW

File: src/app/core/http/authentication-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return this.currentUserSubject.value;
+  }
+
+  login(loginForm: User) {
+    return this.http.post<any>(`${environment.api_url}/users/authenticate`, { loginForm })
+      .pipe(map(user => {
+        // login successful if there's a jwt token in the response
+        if (user && user.token) {
+          // store user details and jwt token in local storage to keep user logged in between page refreshes
+          localStorage.setItem('currentUser', JSON.stringify(user));
```
Comment (bogdanconstantinescu) [inline src/app/core/http/authentication-api.service.ts]: I would create a service that handles the localStorage things for consistency reasons.

File: src/app/core/http/authentication-api.service.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      }));
+  }
+
+  sigup(signupForm: User) {
+    return this.http.post<any>(`${environment.api_url}/users/register`, { signupForm });
+  }
+
+  logout() {
+    // remove user from local storage to log user out
+    localStorage.removeItem('currentUser');
```
Comment (bogdanconstantinescu) [inline src/app/core/http/authentication-api.service.ts]: Same as above, I'd add the localStorage part to a service

**Comment by Wydder** [comment]
I pushed the changes made from review in FileStructure branch

**Comment by RaduCStefanescu** [comment]
Closing this one as the contents are already merged in https://github.com/code4romania/legal-consultation-client/pull/13

**Pull request closed without merging**